app-creator/data/templates/templates_test.go

```diff
@@ -0,0 +1,149 @@
+package templates_data
+
+import (
+	"testing"
+	"net/http/httptest"
+	"io/ioutil"
+	"net/http"
+	"encoding/json"
+	"strings"
+)
+
+func TestToJSON(t *testing.T) {
+	rec := httptest.NewRecorder()
+
+	err := templates.ToJSON(rec)
+	if err != nil {
+		t.Fatalf("error converting to json: %v", err)
+	}
+
+	res := rec.Result()
+	defer res.Body.Close()
+	if res.StatusCode != http.StatusOK {
+		t.Fatalf("expected status OK; got %v", res.StatusCode)
```

Comment:
Got before want in error messages (https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures)